docs/code-samples/getting-started/daml/Scripts/Holding.daml

```diff
@@ -0,0 +1,145 @@
+    bob : Party
+    public : Party
+    aliceAccount : AccountKey
+    bobAccount : AccountKey
+    cashInstrument : InstrumentKey
+    holdingFactoryCid : ContractId Holding.F
+    aliceCashHoldingCid : ContractId Holding.I
+  deriving (Eq, Show)
+
+-- | Test script that
+-- 1. creates an account for Alice and Bob at the Bank
+-- 2. issues a cash instrument
+-- 3. credits a cash holding to Alice in her bank account
+-- 4. transfers the holding from Alice to Bob
+runHolding : Script HoldingState
+runHolding = do
+
+  -- Allocate parties
+  [alice, bank, bob, public] <- mapA createParty $ ["Alice", "Bank", "Bob", "Public"]
+
+  -- Account Factory (it is used by the bank to create accounts)
+  -- CREATE_ACCOUNT_FACTORY_BEGIN
+  accountFactoryCid <- toInterfaceContractId @Account.F <$> submit bank do
+    createCmd Account.Factory with provider = bank; observers = empty
+  -- CREATE_ACCOUNT_FACTORY_END
+
+  -- Holding Factory (it is used by the bank to create holdings with the desired implementation)
+  -- CREATE_HOLDING_FACTORY_BEGIN
+  holdingFactoryCid <- toInterfaceContractId @Holding.F <$> submit bank do
+    createCmd Fungible.Factory with
+      provider = bank
+      observers = fromList [("PublicObserver", singleton public )]
+  -- CREATE_HOLDING_FACTORY_END
+
+  -- Alice and Bob setup account @Bank
+  -- SETUP_ALICE_ACCOUNT_BEGIN
+  aliceRequestCid <- submit alice do
+    createCmd CreateAccount.Request with owner = alice; custodian = bank
+  aliceAccount <- submit bank do
+    exerciseCmd aliceRequestCid CreateAccount.Accept with
+      label = "Alice@Bank"
+      description = "Account of Alice at Bank"
+      accountFactoryCid = accountFactoryCid
+      holdingFactoryCid = holdingFactoryCid
+      observers = []
+  -- SETUP_ALICE_ACCOUNT_END
+
+  bobRequestCid <- submit bob do createCmd CreateAccount.Request with owner = bob; custodian = bank
+  bobAccount <- submit bank do
+    exerciseCmd bobRequestCid CreateAccount.Accept with
+      label = "Bob@Bank"
+      description = "Account of Bob at Bank"
+      accountFactoryCid = accountFactoryCid
+      holdingFactoryCid = holdingFactoryCid
+      observers = [alice]
+
+  -- Bank creates the cash instrument
+  -- ISSUE_CASH_INSTRUMENT_BEGIN
+  let
+    instrumentId = Id "USD"
+    instrumentVersion = "0"
+    instrumentKey = InstrumentKey with
+      issuer = bank
+      depository = bank
+      id = instrumentId
+      version = instrumentVersion
+  now <- getTime
+
+  cashInstrumentCid <- toInterfaceContractId @Instrument.I <$> submit bank do
+    createCmd Instrument with
+      depository = bank
+      issuer = bank
+      id = instrumentId
+      version = instrumentVersion
+      description = "Instrument representing units of USD"
+      validAsOf = now
+      observers = empty
+  -- ISSUE_CASH_INSTRUMENT_END
+
+  -- Alice and Bank create a holding directly using the template
+  -- CREATE_ALICE_DIRECT_HOLDING_BEGIN
+  aliceHoldingCid <- submitMulti [alice, bank] [] do
```

Comment:
Perhaps we should credit the holding without have to do a `submitMulti [alice, bank]` ?

Reply:
I replaced this with a propose/accept now.